client/lib/query-args/index.ts

```diff
@@ -0,0 +1,74 @@
+import { getQueryArgs as wpGetQueryArgs } from '@wordpress/url';
+import page from 'page';
+
+function getRelativeUrlWithParameters(
+	queryArgs: object,
+	clearExistingParameters = false
+): string {
+	const url = new URL( window.location.href );
+
+	if ( clearExistingParameters ) {
+		url.searchParams.forEach( ( value, key ) => url.searchParams.delete( key ) );
```

Comment:
Slight issue with this approach, follow up issue for fix here https://github.com/Automattic/wp-calypso/issues/64100